src/main/scala/co/teapot/tempest/server/DatabaseConfig.scala

```diff
@@ -1,41 +1,19 @@
-/*
- * Copyright 2016 Teapot, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
- * file except in compliance with the License. You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software distributed
- * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
- * CONDITIONS OF ANY KIND, either express or implied. See the License for the
- * specific language governing permissions and limitations under the License.
- */
-
 package co.teapot.tempest.server
-
 import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
 
-import scala.beans.BeanProperty
+/**
+  * Created by gkk on 6/14/17.
+  */
+trait DatabaseConfig {
 
-class DatabaseConfig {
-  @BeanProperty var host: String = ""
-  @BeanProperty var database: String = "tempest"
-  @BeanProperty var user: String = "tempest"
-  @BeanProperty var password: String = ""
+  def createConnectionSource(): HikariDataSource
 
-  @BeanProperty var databaseCachingRam: String = null
+}
 
-  def createConnectionSource(): HikariDataSource = {
+class H2DatabaseConfig extends DatabaseConfig {
```

Comment:
Is there a reason H2DatabaseConfig is in DatabaseConfig.scala, while PostgresDatabaseConfig is in its own file PostgresDatabaseConfig.scala?  I'm wondering if they should each get their own file, or both be in DatabaseConfig.scala.